File: ocmsdk/src/main/java/com/gigigo/orchextra/core/domain/entities/contentdata/ContentItemTypeLayout.java
package com.gigigo.orchextra.core.domain.entities.contentdata;

public enum ContentItemTypeLayout {
  GRID("grid"),
  FULLSCREEN("fullScreen"),
  CAROUSEL("carousel");

  private final String type;

  ContentItemTypeLayout(String type) {
    this.type = type;
  }

  public String getType() {
    return type;
  }

  public static ContentItemTypeLayout convertStringToEnum(String type) {
    ContentItemTypeLayout[] values = ContentItemTypeLayout.values();
    for (ContentItemTypeLayout value : values) {
      if (value.getType().equals(type)) {
        return value;
      }
    }
    return GRID;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/domain/entities/contentdata/ContentItemTypeLayout.kt
package com.gigigo.orchextra.core.domain.entities.contentdata

enum class ContentItemTypeLayout constructor(val type: String) {
  GRID("grid"),
  CAROUSEL("carousel");

  companion object {

    fun convertFromString(type: String?): ContentItemTypeLayout {
      val values = ContentItemTypeLayout.values()
      for (value in values) {
        if (value.type.equals(type,true)) {
          return value
        }
      }
      return GRID
    }
  }
}